extensions/escape-game/standalone.mjs

```diff
@@ -161,31 +171,59 @@ class Standalone {
     if (typeof duration !== "number") {
       return false;
     }
-
-    ++this.doorOpenPressedCount;
-    await this.helper.broadcastCellSetUpdateToAllUser(
-      'update',
-      "world1",
-      CellSet.fromObject({
-        name: LAYER_DOOR.layerName,
-        cells: []
-      }),
-    );
-    setTimeout(async function(obj) {
-      if (--obj.doorOpenPressedCount == 0) {        
-        await obj.helper.broadcastCellSetUpdateToAllUser(
+    const mapName = "world1";
+    if (++this.doorOpenPressedCount === 1) {  
+      console.log("open!");
+      for (const doorCell of this.doorCells) {
+        await this.helper.broadcastCellSetUpdateToAllUser(
           'update',
-          "world1",
+          mapName,
           CellSet.fromObject({
-            name: LAYER_DOOR.layerName,
-            cells: obj.doorCells
+            name: `DOOR_${doorCell["idx"]}`,
+            cells: [doorCell]
           }),
-        );  
+        );
+      }
+    }
+    setTimeout(async function(obj) {
+      if (--obj.doorOpenPressedCount === 0) {
+        console.log("close!");
+        for (const doorCell of obj.doorCells) {
+          await obj.helper.broadcastCellSetUpdateToAllUser(
+            'update',
+            mapName,
+            CellSet.fromObject({
+              name: `DOOR_${doorCell["idx"]}`,
+              cells: [],
+            }),
+          );
+        }
       }
     }, duration, this);
     return true;
   }
 
+  async onPlayerUpdate(msg) {
+    const {mapCoord, playerID} = msg;
+
+    if (typeof playerID !== 'string') {
+      console.warn('Invalid PlayerSyncMessage with non-string msg.playerID', msg, msg.playerID);
+      return;
+    }
+    if (mapCoord === 'undefined') {
+      // No location? Just skip.
+      return;
+    }
+    const explodeCell = this.helper.gameMap.getCell(mapCoord, LAYER_DOOR.layerName);
+    if (explodeCell) { // if walk into opening door => teleport to somewhere
+      const target = mapCoord.copy();
+      target.x = this.doorTarget['x'];
```

Comment:
You'll need to set the world name/map name, or the player will end up in the wrong place.